extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs

```diff
@@ -0,0 +1,131 @@
+        };
+
+        private async Task<object?> ToTargetType(Type targetType, string connectionString, string containerName, string blobName) => targetType switch
+        {
+            Type _ when targetType == typeof(String) => await GetBlobString(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Stream) => await GetBlobStream(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(Byte[]) => await GetBlobBinaryData(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlobClient) => CreateBlobReference<BlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlockBlobClient) => CreateBlobReference<BlockBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(PageBlobClient) => CreateBlobReference<PageBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(AppendBlobClient) => CreateBlobReference<AppendBlobClient>(connectionString, containerName, blobName),
+            Type _ when targetType == typeof(BlobContainerClient) => CreateBlobContainerClient(connectionString, containerName),
+            _ => await DeserializeToTargetObject(targetType, connectionString, containerName, blobName)
+        };
+
+        private async Task<object?> DeserializeToTargetObject(Type targetType, string connectionString, string containerName, string blobName)
+        {
+            var content = await GetBlobString(connectionString, containerName, blobName);
+            return JsonConvert.DeserializeObject(content, targetType);
+        }
+
+        private async Task<string> GetBlobString(string connectionString, string containerName, string blobName)
+        {
+            var client = CreateBlobReference<BlobClient>(connectionString, containerName, blobName);
+            var download = await client.DownloadContentAsync();
+            return download.Value.Content.ToString();
+        }
+
+        private async Task<Byte[]> GetBlobBinaryData(string connectionString, string containerName, string blobName)
+        {
+            var stream = new MemoryStream();
+            var client = CreateBlobReference<BlobClient>(connectionString, containerName, blobName);
+            await client.DownloadToAsync(stream);
+            return stream.ToArray();
+        }
+
+        private async Task<Stream> GetBlobStream(string connectionString, string containerName, string blobName)
+        {
+            var client = CreateBlobReference<BlobClient>(connectionString, containerName, blobName);
+            var download = await client.DownloadStreamingAsync();
+            return download.Value.Content;
+        }
+
+        private BlobContainerClient CreateBlobContainerClient(string connectionString, string containerName)
+        {
+            BlobContainerClient container = new(connectionString, containerName);
```

Comment:
Does this need to support AAD authentication?

Reply:
We do eventually but AAD is out-of-scope for public preview, @surgupta-msft has been investigating this area